Xamarin.Essentials/Connectivity/Connectivity.uwp.cs

```diff
@@ -15,6 +16,18 @@ public static partial class Connectivity
         static void StopListeners() =>
              NetworkInformation.NetworkStatusChanged -= NetworkStatusChanged;
 
+        static SignalStrength PlatformSignalStrength()
+        {
+            var signalStrength = NetworkInformation.GetConnectionProfiles()
+                .Where(profile => profile != null && profile.IsWlanConnectionProfile)
+                .Select(profile => profile.GetSignalBars())
+                .Where(signalBar => signalBar.HasValue)
+                .OrderBy(strength => strength).FirstOrDefault();
+
+            // Adding 1 to return value because 0 is Unknown and 1 is None. If we have a connection we receive at least 1 in return value
+            return signalStrength.HasValue ? (SignalStrength)signalStrength.Value + 1 : SignalStrength.Unknown;
```

Comment:
According to the [docs](https://docs.microsoft.com/en-us/uwp/api/windows.networking.connectivity.connectionprofile.getsignalbars#Windows_Networking_Connectivity_ConnectionProfile_GetSignalBars) the range returned from the platform is **0 - 5**, we need to handle those cases since the enum is basically 1-4.

I am thinking: 0 = none, 1 = weak, 2-3 = fair, 4-5 = strong

Reply:
Thank you for taking the time to look this up, as it was still a WIP i did not get to sift through all the docs and was not able to test very much 👍 I changed it to your spec.